flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/config/AWSConfigConstants.java

```diff
@@ -154,6 +154,77 @@ public enum CredentialProvider {
 
+    /**
+     * The type of {@link software.amazon.awssdk.http.SdkHttpClient}. If set, all AWS clients will
+     * use this specified HTTP client. If not set, HTTP_CLIENT_TYPE_DEFAULT will be used. For
+     * specific types supported, see HTTP_CLIENT_TYPE_* defined below.
+     */
+    public static final String HTTP_CLIENT_TYPE = "http-client.type";
+
+    // ---- glue configs
+
+    /**
+     * Used to configure the connection timeout in milliseconds for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see <a
+     * href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html">...</a>
+     */
+    public static final String HTTP_CLIENT_CONNECTION_TIMEOUT_MS =
+            "http-client.connection-timeout-ms";
+
+    /**
+     * Used to configure the max connections number for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see <a
+     * href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html">...</a>
+     */
+    public static final String HTTP_CLIENT_APACHE_MAX_CONNECTIONS =
+            "http-client.apache.max-connections";
+
+    /**
+     * Used to configure the socket timeout in milliseconds for {@link
+     * software.amazon.awssdk.http.apache.ApacheHttpClient.Builder}. This flag only works when
+     * {@link #HTTP_CLIENT_TYPE} is set to HTTP_CLIENT_TYPE_APACHE
+     *
+     * <p>For more details, see <a
+     * href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html">...</a>
+     */
+    public static final String HTTP_CLIENT_SOCKET_TIMEOUT_MS = "http-client.socket-timeout-ms";
+
+    /**
+     * Configure an alternative endpoint of the Glue service for GlueCatalog to access.
+     *
+     * <p>This could be used to use GlueCatalog with any glue-compatible metastore service that has
+     * a different endpoint
+     */
+    public static final String GLUE_CATALOG_ENDPOINT = "aws.glue.endpoint";
+
+    /**
+     * The ID of the Glue Data Catalog where the tables reside. If none is provided, Glue
+     * automatically uses the caller's AWS account ID by default.
+     *
+     * <p>For more details, see <a
+     * href="https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-databases.html">...</a>
+     */
+    public static final String GLUE_CATALOG_ID = "aws.glue.id";
+
+    /**
+     * The account ID used in a Glue resource ARN, e.g.
+     * arn:aws:glue:us-east-1:1000000000000:table/db1/table1
+     */
+    public static final String GLUE_ACCOUNT_ID = "aws.glue.account-id";
+
```

Comment:
These are the only configs that are glue specific. We have kept this package abstract as possible, we can move those to the glue catalog package